File: aoj/volume100/csharp/A10026.cs
using System;
using System.Collections.Generic;
using System.Linq;

class AOJ
{
	static void Main()
	{
		while (true)
		{
			var noItem = int.Parse(Console.ReadLine());
			if (noItem == 0)
				break;

			var nums = Console.ReadLine()
				.Split()
				.Select(s => double.Parse(s))
				.ToList();

			Console.WriteLine(StandardDeviation(nums).ToString("f6"));
		}
	}

	static double StandardDeviation(List<double> nums)
	{
		return Math.Sqrt(Variance(nums));
	}

	static double Variance(List<double> nums)
	{
		var avg = Avarage(nums);
		var vsum = 0.0;
		foreach (var n in nums)
			vsum += Math.Pow(n-avg, 2);
		return vsum / nums.Count;
	}

	static double Avarage(List<double> nums)
	{
		var sum = 0.0;
		foreach (var n in nums)
			sum += n;
		return sum / nums.Count;
	}
}

File: aoj/volume100/php/A10026.php
<?php

function average($nums) {
    $sum = 0;

    $len = count($nums);
    for ($i = 0; $i < $len; $i++)
        $sum += $nums[$i];

    return $sum / $len;
}

function variance($nums) {
    $v = 0;
    $average = average($nums);

    $len = count($nums);
    for ($i = 0; $i < $len; $i++)
        $v += pow($nums[$i] - $average, 2);

    return $v / $len;
}

function standard_deviation($nums) {
    return sqrt(variance($nums));
}

while (true) {
    fscanf(STDIN, "%d", $noitem);
    if ($noitem == 0)
        break;

    $nums = array_map("intval", explode(" ", trim(fgets(STDIN))));
    $standard_deviation = standard_deviation($nums);

    printf("%.6f\n", $standard_deviation);
}
